core/build.gradle

```diff
@@ -57,6 +57,7 @@ dependencies {
     api group: 'com.google.code.gson', name: 'gson', version: '2.8.9'
     api group: 'com.tdunning', name: 't-digest', version: '3.3'
     api project(':common')
+    implementation "com.github.seancfoley:ipaddress:5.4.2"
```

Comment:
This is the library I initially wanted to add. Turns out it's already a dependency for `:core` (see `common/build.gradle`), and it was also used for the Spark implementation, so I've added it here so that the behaviour will be the same.